### PR TITLE
Autobuild: Combine and simplify iOS build scripts

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**

- Extract the non-autobuild-related parts into an ios/deploy_ios.sh script, which can run standalone, similar to other platforms (by @ann0see)

- Move all autobuild/ios/* scripts to a single .github/autobuild/ios.sh script which is called for the different stages (setup/build/get-artifacts).

- Condense redundant parameter parsing into a single step

- Create functions with proper names for larger steps

- Inline Github artifact output definition as it's shorter that way

- Make shellcheck-clean

CHANGELOG: Autobuild: Refactored and simplified iOS build scripts.
(will be condensed...)

**Context: Fixes an issue?**

Related: #2503

**Does this change need documentation? What needs to be documented and how?**

**Status of this Pull Request**

Ready for review.

**What is missing until this pull request can be merged?**

Reviews.

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I tested my code and it does what I want: partially
- [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I've filled all the content above

### CODE REVIEW

**Review by ann0see — APPROVED** [review]

File: ios/deploy_ios.sh [hunk 1]
```diff
@@ -1,27 +1,19 @@
...
-cd "${THIS_JAMULUS_PROJECT_PATH}"
+## Builds an ipa file for iOS. Should be run from the repo-root
 
+# Create Xcode file and build
 qmake -spec macx-xcode Jamulus.pro
 /usr/bin/xcodebuild -project Jamulus.xcodeproj -scheme Jamulus -configuration Release clean archive -archivePath "build/Jamulus.xcarchive" CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO
-# Make a deploy folder
-mkdir deploy
+
 # Generate ipa by copying the .app file from the xcarchive file
```
Comment (pljones) [inline ios/deploy_ios.sh:10]: `xcarchive file` looks like a directory.

File: ios/deploy_ios.sh [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
-mkdir deploy/Payload
-cp -r build/Jamulus.xcarchive/Products/Applications/Jamulus.app deploy/Payload/
-cd deploy
+mkdir build/Payload
+cp -r build/Jamulus.xcarchive/Products/Applications/Jamulus.app build/Payload/
+cd build || exit 1
```
Comment (pljones) [inline ios/deploy_ios.sh:13]: `set -eu` should mean the `|| exit 1` isn't needed -- it's effectively the same thing, as far as I'm aware.

Comment (hoffie) [reply ios/deploy_ios.sh]: Hmm, [shellcheck made me do this](https://github.com/koalaman/shellcheck/wiki/SC2164) and I assumed that `cd` (as a shell built-in) was special compared to other commands which I know are checked with `set -e`.
I've run a quick test and it looks you are right. I'm going to remove the `|| exit 1`.

This is what happened (probably): I took the script, ran shell check, noted the warning, fixed it and only later noticed that the script did not use `set -eu` and added that. So, rather PEBKAC than a shellcheck bug... :)

File: ios/deploy_ios.sh (hunk 1 above)
Comment (hoffie) [reply ios/deploy_ios.sh]: I don't know anything about that part, but based on the `cp` invocation, it must be a directory, so I've adjusted the comment as well (cc @ann0see).

Comment (ann0see) [reply ios/deploy_ios.sh]: On macOS it displays as a file although you can cd into it. It’s similar to .app files. They show up as file, but are essentially folders

Comment (pljones) [reply ios/deploy_ios.sh]: A truely broken design...

**Review by pljones — APPROVED** [review]
Looks very nice now.

**Pull request merged**